docs/architecture/adr-042-group-module.md

```diff
@@ -0,0 +1,375 @@
+those "sub-accounts" using the [`x/authz` module](adr-030-authz-module.md).
+
+Group accounts are stored as part of the `groupAccountTable` and modeled by `GroupAccountInfo`.
+
+```proto
+message GroupAccountInfo {
+
+    // address is the group account address.
+    string address = 1;
+    
+    // group_id is the unique ID of the group.
+    uint64 group_id = 2;
+
+    // admin is the account address of the group admin.
+    string admin = 3;
+    
+    // metadata is any arbitrary metadata to attached to the group account.
+    bytes metadata = 4;
+
+    // version is used to track changes to a group's GroupAccountInfo structure that
+    // would create a different result on a running proposal.
+    uint64 version = 5;
+
+    // decision_policy specifies the group account's decision policy.
+    google.protobuf.Any decision_policy = 6 [(cosmos_proto.accepts_interface) = "DecisionPolicy"];
+}
+```
+
+The group account address is generated based on an auto-increment integer which is used to derive the group module `RootModuleKey` into a `DerivedModuleKey`, as stated in [ADR-033](adr-033-protobuf-inter-module-comm.md#modulekeys-and-moduleids). The group account is added as a new `ModuleAccount` through `x/auth`.
+
+### Decision Policy
+
+A decision policy is the mechanism by which members of a group can vote on 
+proposals.
+
+All decision policies generally would have a minimum and maximum voting window.
+The minimum voting window is the minimum amount of time that must pass in order
+for a proposal to potentially pass, and it may be set to 0. The maximum voting
+window is the maximum time that a proposal may be voted on before it is closed.
+Both of these values must be less than a chain-wide max voting window parameter.
+
+We define the `DecisionPolicy` interface that all decision policies must implement:
+
+```go
+type DecisionPolicy interface {
+	codec.ProtoMarshaler
+
+	ValidateBasic() error
+	GetTimeout() types.Duration
+	Allow(tally Tally, totalPower string, votingDuration time.Duration) (DecisionPolicyResult, error)
+	Validate(g GroupInfo) error
+}
+
+type DecisionPolicyResult struct {
+	Allow bool
+	Final bool
+}
+```
+
+#### Threshold decision policy
+
+A threshold decision policy defines a threshold of yes votes (based on a tally
+of voter weights) that must be achieved in order for a proposal to pass. For
+this decision policy, abstain and veto are simply treated as no's.
+
+```proto
+message ThresholdDecisionPolicy {
+
+    // threshold is the minimum weighted sum of yes votes that must be met or exceeded for a proposal to succeed.
+    string threshold = 1;
+    
+    // timeout is the duration from submission of a proposal to the end of voting period
+    // Within this times votes and exec messages can be submitted.
+    google.protobuf.Duration timeout = 2 [(gogoproto.nullable) = false];
+}
+```
+
+### Proposal
+
+Any member of a group can submit a proposal for a group account to decide upon.
```

Comment:
```suggestion
Any member of a group can submit a proposal to a group account to vote and execute.
```

Reply:
This is not entirely true, anyone can execute a proposal.